src/main/java/bacnet/BacnetIpConnection.java

```diff
@@ -0,0 +1,157 @@
+package bacnet;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.Map;
+
+import org.dsa.iot.dslink.node.Node;
+import org.dsa.iot.dslink.node.actions.Action;
+import org.dsa.iot.dslink.node.actions.ActionResult;
+import org.dsa.iot.dslink.node.actions.Parameter;
+import org.dsa.iot.dslink.node.value.Value;
+import org.dsa.iot.dslink.node.value.ValueType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.serotonin.bacnet4j.exception.BACnetException;
+import com.serotonin.bacnet4j.npdu.Network;
+import com.serotonin.bacnet4j.npdu.ip.IpNetwork;
+import com.serotonin.bacnet4j.npdu.ip.IpNetworkBuilder;
+import com.serotonin.bacnet4j.npdu.ip.IpNetworkUtils;
+import com.serotonin.bacnet4j.type.primitive.OctetString;
+
+public class BacnetIpConnection extends BacnetConn {
+	private static final Logger LOGGER;
+
+	static {
+		LOGGER = LoggerFactory.getLogger(BacnetConn.class);
+	}
+
+	static final String ATTR_BROADCAST_IP = "broadcast ip";
+	static final String ATTR_PORT = "port";
+	static final String ATTR_LOCAL_BIND_ADDRESS = "local bind address";
+	static final String ATTR_REGISTER_AS_FOREIGN_DEVICE = "register as foreign device in bbmd";
+	static final String ATTR_BBMD_IP_WITH_NETWORK_NUMBER = "bbmd ips with network number";
+
+	String broadcastIp;
+	int port;
+	String localBindAddress;
+	boolean isRegisteredAsForegnDevice;
+	String bbmdIpList;
+
+	BacnetIpConnection(BacnetLink link, Node node) {
+		super(link, node);
+
+		broadcastIp = node.getAttribute(ATTR_BROADCAST_IP).getString();
+		port = node.getAttribute(ATTR_PORT).getNumber().intValue();
+		localBindAddress = node.getAttribute(ATTR_LOCAL_BIND_ADDRESS).getString();
+		isRegisteredAsForegnDevice = node.getAttribute(ATTR_REGISTER_AS_FOREIGN_DEVICE).getBool();
+		bbmdIpList = node.getAttribute(ATTR_BBMD_IP_WITH_NETWORK_NUMBER).getString();
+		if (null != bbmdIpList && isRegisteredAsForegnDevice) {
+			this.parseBBMD(bbmdIpList);
+		}
+	}
+
+	Network createNetwork(int localNetworkNumber) {
+		IpNetworkBuilder builder = new IpNetworkBuilder();
+		Network network = builder.broadcastIp(broadcastIp).port(port).localBindAddress(localBindAddress)
+				.localNetworkNumber(localNetworkNumber).build();
+		return network;
+	}
+
+	private void parseBBMD(String bbmdIpList) {
```

Comment:
bbmd = broadcast management device. Usually the short name is used in the bacnet protocol.